src/sst/elements/memHierarchy/cacheController.cc

```diff
@@ -356,6 +356,7 @@ bool Cache::allocateLine(MemEvent * event, Addr baseAddr) {
             return false;
         }
         
+        notifyListenerOfEvict(event, replacementLine);
```

Comment:
Move this call a few lines down to the actual array->replace() position since the eviction can fail.